lib/profiler/get_tick.h

```diff
@@ -67,6 +67,19 @@ get_tick(void)
     return regs;
 }
 
+#elif defined(__APPLE__)
+#include <mach/mach_time.h>
+
+static __inline hrtime_t
+get_tick(void)
+{
+    static mach_timebase_info_data_t base;
+    uint64_t ticks;
+    (void) mach_timebase_info (&base);
```

Comment:
```
Prefer to use the equivalent clock_gettime_nsec_np(CLOCK_UPTIME_RAW) in nanoseconds.
```

Apple [recommends](https://developer.apple.com/documentation/kernel/1462446-mach_absolute_time) using clock_gettime_nsec_np() instead. If that recommendation is wrong or does not apply, please add a comment to explain why we disregard it. Otherwise, please follow that recommendation.

Reply:
Right that was because mach_timebase_info is the historical api but in the case of the M1 indeed makes less sense since it s brand new anyway.